controllers/secret.go

```diff
@@ -181,6 +181,23 @@ func (r *OperatorPipelineReconciler) reconcilePyxisAPISecret(meta metav1.ObjectM
 	return nil // Existing Secret found, do nothing...
 }
 
+// dockerRegistrySecretExists will check if the Docker Registry Secret is present.
+func (r *OperatorPipelineReconciler) secretExists(meta metav1.ObjectMeta, secret_purpose string, secret_name string) bool {
```

Comment:
I believe Sid's PR to refactor secrets could be used here.

Reply:
great point. will wait for that PR and refactor